### PR TITLE
Bug 1870339 - Make list of add-ons more accessible

### PR DESCRIPTION
**Pull request by willdurand**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1870339

### CODE REVIEW

**Comment by willdurand** [comment]
<table>
<tr>
<td>

signal-2024-02-15-124750

</td>
<td>

signal-2024-02-15-124812

</td>
</tr>
</table>

**Comment by willdurand** [comment]
@AaronMT hey, sorry to bother you but would you have an idea why some UI tests are now failing?

File: android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt [hunk 1]
```diff
@@ -181,9 +183,35 @@ class AddonsManagerAdapter(
...
         val item = getItem(position)
 
+        // Configure an accessibility delegate for each item.
+        holder.itemView.accessibilityDelegate = object : View.AccessibilityDelegate() {
+            override fun onInitializeAccessibilityNodeInfo(host: View, info: AccessibilityNodeInfo) {
+                super.onInitializeAccessibilityNodeInfo(host, info)
+
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                    info.collectionItemInfo = AccessibilityNodeInfo.CollectionItemInfo(
+                        position,
```
Comment (Amejia481) [inline android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt:196]: I was taking a look at the`@SuppressLint("RecyclerView")` , it seems that we shouldn't be using the `position` parameter, instead we should be using `getAdapterPosition()` , bellow the reference:


> Do not treat position as fixed; only use immediately and call holder.getAdapterPosition() to look it up later More... (⌘F1)  Inspection info: RecyclerView will not call onBindViewHolder again when the position of the item changes in the data set unless the item itself is invalidated or the new position cannot be determined.  For this reason, you should only use the position parameter while acquiring the related data item inside this method, and should not keep a copy of it.  If you need the position of an item later on (e.g. in a click listener), use getAdapterPosition() which will have the updated adapter position.  Issue id: RecyclerView Vendor: Android Open Source Project Contact: https://groups.google.com/g/lint-dev Feedback: https://issuetracker.google.com/issues/new?component=192708

**Comment by Amejia481** [comment]
> @AaronMT hey, sorry to bother you but would you have an idea why some UI tests are now failing?

Looking at the stack trace, looks like we are not able to find the `addonName` on the list.

https://github.com/mozilla-mobile/firefox-android/blob/d3df899480e59b73ae9d0393b340db9b7267ba91/fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt#L92

`org.mozilla.fenix.ui.SettingsAddonsTest.installAddonFromMainMenuTest`

```
androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[SCROLLABLE=true]
	at androidx.test.uiautomator.UiScrollable.scrollBackward(UiScrollable.java:496)
	at androidx.test.uiautomator.UiScrollable.scrollToBeginning(UiScrollable.java:544)
	at androidx.test.uiautomator.UiScrollable.scrollToBeginning(UiScrollable.java:563)
	at androidx.test.uiautomator.UiScrollable.scrollIntoView(UiScrollable.java:265)
	at androidx.test.uiautomator.UiScrollable.scrollTextIntoView(UiScrollable.java:336)
	at org.mozilla.fenix.helpers.TestHelper.scrollToElementByText(TestHelper.kt:57)
	at org.mozilla.fenix.ui.robots.SettingsSubMenuAddonsManagerRobot.verifyAddonIsInstalled(SettingsSubMenuAddonsManagerRobot.kt:141)
	at org.mozilla.fenix.ui.SettingsAddonsTest$installAddonFromMainMenuTest$3.invoke(SettingsAddonsTest.kt:92)
	at org.mozilla.fenix.ui.SettingsAddonsTest$installAddonFromMainMenuTest$3.invoke(SettingsAddonsTest.kt:75)
	at org.mozilla.fenix.ui.robots.ThreeDotMenuMainRobot$Transition.openAddonsManagerMenu(ThreeDotMenuMainRobot.kt:451)
	at org.mozilla.fenix.ui.SettingsAddonsTest.installAddonFromMainMenuTest(SettingsAddonsTest.kt:75)
```

**Comment by AaronMT** [comment]
> @AaronMT hey, sorry to bother you but would you have an idea why some UI tests are now failing?

Sorry I missed this, it's best to ping `@mobiletest` in Slack if you are blocked here. I'll loop in @sv-ohorvath who is more familiar with this test, but I think @Amejia481 above identified the problem.

**Comment by AndiAJ** [comment]
Hi, it looks like for unknown reasons the scrolling action to an add-on wasn't properly performed `scrollToElementByText(addonName)`

I've made a fix [here](https://github.com/mozilla-mobile/firefox-android/pull/5717/commits/f7a66e60525f0e77ee2dcc1f09673583d2b44d62#diff-8e8c3790ab858c3fe796c7b006f564e8ebdcf6bb9e337f8d8e2365450a1b1f4e) and seems to be working (all add-ons related UI tests successfully passed 25x on Firebase)

I'll send the patch on slack to @willdurand ☺️

File: android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt (hunk 1 above)
Comment (willdurand) [reply android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt:196]: We don't do this in the codebase elsewhere so I dunno

Comment (willdurand) [reply android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt:197]: updated

**Review by Amejia481 — APPROVED** [review]
✅

**Pull request merged**